File: ILogReader.py
class ILogReader:
    def read_log_file(self, identificator: str) -> str:
        pass

class FtpLogReader(ILogReader):
    def __init__(self, login: str, password: str):
        self.login = login
        self.password = password
        self.ftp_client = FtpClient()

    def read_log_file(self, file_path: str) -> str:
        return self.ftp_client.read_file(self.login, self.password, file_path)

File: ILogReader.cs
using Patterns.Ex00.ExternalLibs;

namespace Patterns.Ex00
{
    public class FtpLogReader : ILogReader
    {
        private readonly string _login;
        private readonly string _password;
        private readonly FtpClient _ftpClient;

        public FtpLogReader(string login, string password)
        {
            _login = login;
            _password = password;
            _ftpClient = new FtpClient();
        }

        public string ReadLogFile(string filePath)
        {
            return _ftpClient.ReadFile(_login, _password, filePath);
        }
    }
}